.eslintrc

```diff
@@ -49,32 +58,60 @@
         "no-unused-expressions": 2,
         "no-with": 2,
         "radix": 2,
+        "vars-on-top": 1,
```

Comment:
Totally agree, it may be a bit nagging. On the bright side, I think it also forces good naming of variables and makes it easier to add inline comments about the purpose of more opaque ones. 

It very well reminds us of variable hoisting as well. Though that has been partially taken care of by the `no-use-before-define` setting already.